server/src/main/java/org/opensearch/action/admin/cluster/node/hotthreads/NodesHotThreadsRequest.java

```diff
@@ -71,6 +71,7 @@ public NodesHotThreadsRequest(StreamInput in) throws IOException {
      */
     public NodesHotThreadsRequest(String... nodesIds) {
         super(nodesIds);
+        this.setIncludeDiscoveryNodes(false);
```

Comment:
you are setting this explicitly as plugin can extend TransportNodesAction, trying to understand what if we would have set it false in TransportNodesAction itself

Reply:
Without a flag, I think we will end up not setting the `nodes` for transport requests made by plugin also. Are we thinking on having a flag in the TransportNodesAction and reset by derived actions ?